Formula/openlitespeed.rb

```diff
@@ -11,19 +11,83 @@ class Openlitespeed < Formula
     sha256 "aacafddf229c07c76481f3fc86b3295ca06b41e14555b0e8a97050098701a1d2" => :mavericks
   end
 
-  option "with-debug", "Compile with support for debug log"
-  option "with-spdy", "Compile with support for SPDY module"
+  option "with-debug", "Build with support for debug log"
+  option "with-redis", "Enable redis for cache module"
+
+  option "without-http2", "Disable SPDY and http2"
 
   depends_on "pcre"
   depends_on "geoip"
   depends_on "openssl"
+  depends_on "expat"
+  depends_on "zlib"
 
   def install
-    args = ["--disable-dependency-tracking", "--prefix=#{prefix}"]
+    args = %W[
+      --disable-dependency-tracking
+      --prefix=#{prefix}
+      --with-pcre=#{Formula["pcre"].opt_prefix}
+      --with-openssl=#{Formula["openssl"].opt_prefix}
+      --with-zlib=#{Formula["zlib"].opt_prefix}
+      --with-expat=#{Formula["expat"].opt_prefix}
+      --with-expat-inc=#{Formula["expat"].opt_prefix}/include
+      --with-expat-lib=#{Formula["expat"].opt_prefix}/lib
+    ]
+
     args << "--enable-debug" if build.with? "debug"
-    args << "--enable-spdy" if build.with? "spdy"
-    args << "--with-openssl=#{Formula["openssl"].opt_prefix}"
+    args << "--enable-redis=yes" if build.with? "redis"
+    args << "--enable-http2=no" if build.without? "http2"
+
+    ENV["CPPFLAGS"] = "-I#{Formula["openssl"].opt_prefix}/include/openssl"
+
     system "./configure", *args
+    system "make"
     system "make", "install"
+
+    inreplace bin/"lswsctrl.open", "RESTART_LOG=\"\$BASE_DIR\/\.\.\/logs\/lsrestart.log\"", "RESTART_LOG=\"#{opt_prefix}\/logs\/lsrestart.log\""
+  end
+
+  def post_install
+    (prefix/"logs").mkpath
+    (prefix/"admin/cgid").mkpath
+  end
+
+  plist_options :manual => "LSWS_HOME=#{HOMEBREW_PREFIX}/opt/openlitespeed lswsctrl start"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple Computer//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+    <dict>
+      <key>EnvironmentVariables</key>
+      <dict>
+        <key>LSWS_HOME</key>
+        <string>#{prefix}</string>
+      </dict>
+      <key>Label</key>
+      <string>#{plist_name}</string>
+      <key>ProgramArguments</key>
+      <array>
+        <string>#{opt_bin}/lswsctrl</string>
+        <string>start</string>
+      </array>
+      <key>RunAtLoad</key>
+      <true/>
+      <key>StandardErrorPath</key>
+      <string>#{var}/logs/openlitespeed-error.log</string>
+      <key>StandardOutPath</key>
+      <string>#{var}/logs/openlitespeed.log</string>
+    </dict>
+    </plist>
+    EOS
+  end
+
+  test do
```

Comment:
Seems a bit overly complicated. Anything wrong with:

```ruby
ENV["LSWS_HOME"] = prefix
system bin/"lswsctrl", "start"
system bin/"lswsctrl", "status"
system bin/"lswsctrl", "stop"
```
?

Would be good to assert something on the `status` output as well.